Sources/GRPC/ConnectionPool/PoolManager.swift

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright 2021, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Logging
+import NIO
+import NIOConcurrencyHelpers
+
+internal final class PoolManager {
+  /// Configuration used for each connection pool.
+  internal struct PerPoolConfiguration {
+    /// The maximum number of connections per pool.
+    var connections: Int
```

Comment:
Maybe call this `maxConnections`? You need to read the docu to know that this is about the max number...